tests/instantiate.test.ts

```diff
@@ -0,0 +1,49 @@
+import instantiateUuids from '../src';
+jest.mock('uuid', () => ({
+  v4: jest.fn(),
+}));
+
+beforeEach(() => {
+  const mockUuid = require('uuid') as { v4: jest.Mock<string, []> }; // "as" is TypeScript you can ignore
+  mockUuid.v4
+    .mockImplementationOnce(() => 'uuid-1')
+    .mockImplementationOnce(() => 'uuid-2');
+});
+
+afterEach(() => {
+  jest.resetAllMocks();
+});
+
+test('uuid algorithm conversion works', () => {
+  const example = `<div resource="https://example.org/--ref-<uuid>-123">`;
+  const result = instantiateUuids(example);
+  expect(result).toBe(`<div resource="https://example.org/uuid-1">`);
+});
+
+test('memoization works', () => {
+  const example = `
+        <div resource="https://example.org/--ref-<uuid>-123">
```

Comment:
Actually, looking at [an example of this in use](https://github.com/lblod/ember-rdfa-editor-lblod-plugins/pull/440/files#diff-1d08d969d8e0b4df77a2c8c7bf43f91090ecbd9a6ad51adf30a3e7c9c53c2936), I think the intent is for the ref part to be of the form `--ref-uuid4-123`. Come to think of it, since we're saying uuid and uuids are often represented with `-`s, it's worth adding a realistic uuid to the tests to make sure it does parse correctly (I think it does, but to make the tests more thorough).

Reply:
done :+1: